src/ucc/api/ucc.h

```diff
@@ -1574,5 +1588,229 @@ static inline ucc_status_t ucc_collective_test(ucc_coll_req_h request)
  */
 ucc_status_t ucc_collective_finalize(ucc_coll_req_h request);
 
+
+/**
+ * @ingroup UCC_EVENT_DT
+ *
+ */
+
+typedef enum ucc_event_type {
+	UCC_EVENT_COLLECTIVE_POST       = UCC_BIT(0),
+	UCC_EVENT_COLLECTIVE_COMPLETE   = UCC_BIT(1),
+	UCC_EVENT_COMPUTE_COMPLETE      = UCC_BIT(2),
+	UCC_EVENT_OVERFLOW              = UCC_BIT(3)
```

Comment:
do we need to define somewhere that some events ( UCC_EVENT_COLLECTIVE_POST, UCC_EVENT_COLLECTIVE_COMPLETE) are generated by UCC and some events ( UCC_EVENT_COMPUTE_COMPLETE) are set by the user?